server/api/smallgroup/smallgroup.controller.js

```diff
@@ -304,5 +304,20 @@ function generateCode(codeLength){
       var character = (Math.floor(Math.random() * characterOptions.length));
       code = code.concat(characterOptions[character.toString()]);
   }
+  ClassYear.findOne({"dayCodes.code":code})
```

Comment:
This entire segment doesn't currently get executed before the  return code at the bottom is finished. Therefore if it finds the code is duplicate, nothing but an exception would occur.